src/tables.jl

```diff
@@ -0,0 +1,105 @@
+const AG = ArchGDAL
+
+struct GeoTable{T<:Union{IFeatureLayer, FeatureLayer}} 
+    layer::T
+end
+
+function geotable(layer)
+    featuredefn = layerdefn(layer)
+    ngeometries = ngeom(featuredefn)
+    if(ngeometries == 0 )
+        println("NULL Geometry found")
+    end
+    GeoTable(layer)
+end
+
+function Tables.schema(layer::AG.AbstractFeatureLayer)
+    nfield = AG.nfield(layer)
+    featuredefn = AG.layerdefn(layer)
+    ngeom = ArchGDAL.ngeom(featuredefn)
+
+    fielddefns = (AG.getfielddefn(featuredefn, i) for i in 0:nfield-1)
+    names_fields = Tuple(AG.getname(fielddefn) for fielddefn in fielddefns)
+    types_fields = Tuple(AG._FIELDTYPE[AG.gettype(fielddefn)] for fielddefn in fielddefns)
+    
+    Tables.Schema(names_fields, types_fields)
+end
+
+function Base.iterate(gt::GeoTable, st = 0)
+    layer = gt.layer
+    AG.resetreading!(layer)
+    nfeat = AG.nfeature(layer)
+    nfield = AG.nfield(layer)
+    featuredefn = layerdefn(layer)
+
+    d = Dict{String, Vector}()
+
+    for field_no in 0:nfield-1
+        field = getfielddefn(featuredefn, field_no)
+        name = getname(field)
+        d[name] = Vector{_FIELDTYPE[gettype(field)]}()
+    end
+    
+    st >= nfeat && return nothing
+    AG.getfeature(layer, st) do feature
+        for (k, v) in pairs(d)
+            val = getfield(feature, k)
+            push!(v, val)
+        end
+    end
+    
+    keys_tup = ()
+    for _key in keys(d)
+        keys_tup = (keys_tup..., Symbol(_key))
+    end
+    vals_tup = Tuple(values(d))
+    
+    #Using the tables interface
+    Row = Tables.rowtable(NamedTuple{keys_tup}(vals_tup))
```

Comment:
So the main reason of doing this is https://github.com/Sov-trotter/ArchGDAL.jl/blob/dev-tables/src/tables.jl#L40. If we simply return a splatted NamedTuple, then every item is a vector.
![Screenshot from 2020-08-15 15-58-15](https://user-images.githubusercontent.com/43717431/90310570-43e87500-df10-11ea-9607-a44dd74a2083.png)

Reply:
Don't see why it should be a vector though? We should probably get rid of all the dicts and vectors in this method, and just stick to NamedTuples of combined geometries and attributes.